src/frequenz/sdk/actor/power_distributing/_component_pool_status_tracker.py

```diff
@@ -0,0 +1,163 @@
+    """Track status of components of a given category.
+
+    Send set of working and uncertain components, when the status of any of the tracked
+    components changes.
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        component_ids: set[int],
+        component_status_sender: Sender[ComponentPoolStatus],
+        max_data_age_sec: float,
+        max_blocking_duration_sec: float,
+        component_status_tracker_type: type[ComponentStatusTracker],
+    ) -> None:
+        """Create ComponentPoolStatusTracker instance.
+
+        Args:
+            component_ids: set of component ids whose status is to be tracked.
+            component_status_sender: The sender used for sending the status of the
+                tracked components.
+            max_data_age_sec: If a component stops sending data, then this is the
+                maximum time for which its last message should be considered as
+                valid. After that time, the component won't be used until it starts
+                sending data.
+            max_blocking_duration_sec: This value tell what should be the maximum
+                timeout used for blocking failing component.
+            component_status_tracker_type: component status tracker to use
+                for tracking the status of the components.
+        """
+        self._component_ids = component_ids
+        self._max_data_age_sec = max_data_age_sec
+        self._max_blocking_duration_sec = max_blocking_duration_sec
+        self._component_status_sender = component_status_sender
+        self._component_status_tracker_type = component_status_tracker_type
+
+        # At first no component is working, we will get notification when they start
+        # working.
+        self._current_status = ComponentPoolStatus(working=set(), uncertain=set())
+
+        # Channel for sending results of requests to the components.
+        self._set_power_result_channel = Broadcast[SetPowerResult](
+            "component_request_status"
+        )
+        self._set_power_result_sender = self._set_power_result_channel.new_sender()
+        self._component_status_trackers: list[ComponentStatusTracker] = []
+        self._merged_status_receiver = self._make_merged_status_receiver()
+
+        self._task = asyncio.create_task(self._run())
+
+    async def join(self) -> None:
+        """Wait and return when the instance's task completes.
+
+        It will not terminate the instance, which can be done with the `stop` method.
+        """
+        await self._task
+
+    async def stop(self) -> None:
+        """Stop tracking batteries status."""
+        await cancel_and_await(self._task)
+        await asyncio.gather(
+            *[tracker.stop() for tracker in self._component_status_trackers],
+        )
+        await self._merged_status_receiver.stop()
+
+    def _make_merged_status_receiver(
+        self,
+    ) -> Merge[ComponentStatus]:
```

Comment:
It might be better to return a `Receiver` here as `Merge` is becoming private.

Reply:
So we stop calling `stop` on it in tests, etc, right?